cores/esp8266/Arduino.h

```diff
@@ -218,28 +216,31 @@ void optimistic_yield(uint32_t interval_us);
 } // extern "C"
 #endif
 
+// undefine stdlib's definitions when encountered, provide abs that supports floating point for C code
+#ifndef __cplusplus
+#undef abs
+#define abs(x) ((x)>0?(x):-(x))
```

Comment:
For sanity I would
```
#define abs(x) ({ __typeof__(x) _x = (x); _x > 0 ? _x : -_x; })
#define round(x) ({ __typeof__(x) _x = (x); _x >= 0 ? (long)(_x + 0.5) : (long)(_x - 0.5); })
```

Reply:
Why are these defined at all? I think eboot.c and umm is the only C code in our core, and Arduino.h isn't included in either AFAIK.

Reply:
Not the Core, these are for users' .c code